src/Microsoft.DotNet.Interactive.Mermaid/UmlClassDiagram.cs

```diff
@@ -0,0 +1,161 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Reflection;
+using System.Text;
+
+namespace Microsoft.DotNet.Interactive.Mermaid;
+
+public class UmlClassDiagram
+{
+    private readonly Type _type;
+    private UmlClassDiagramConfiguration? _classDiagramConfiguration;
+    private readonly MermaidMarkdown _markDown;
+
+    public UmlClassDiagram(Type type, UmlClassDiagramConfiguration? classDiagramConfiguration)
+    {
+        _type = type;
+        _classDiagramConfiguration = classDiagramConfiguration;
+        _markDown = ToClassDiagram(_type, _classDiagramConfiguration);
+    }
+
+    public MermaidMarkdown ToMarkdown()
+    {
+        return _markDown;
+    }
+
+    public UmlClassDiagram WithGraphDepth(int graphDepth)
+    {
+        _classDiagramConfiguration ??= new UmlClassDiagramConfiguration();
```

Comment:
It'll be more predictable to move the null coalesce to the constructor and make the field `readonly`.